packages/amplify-category-storage/src/provider-utils/awscloudformation/cdk-stack-builder/s3-stack-builder.ts

```diff
@@ -177,6 +177,11 @@ export class AmplifyS3ResourceCfnStack extends AmplifyResourceCfnStack implement
       bucketName: this.s3BucketName,
       corsConfiguration: this.buildCORSRules(),
     });
+    // eslint-disable-next-line no-new
+    new s3Cdk.CfnBucketPolicy(this, 'DeploymentBucketBlockHTTP', {
+      bucket: this.s3BucketName,
+      policyDocument: this.addSecureTransportPolicyDocument(),
+    });
```

Comment:
is this our bucket or customer bucket ?

If this is customer bucket then we should probably provide option to disable this.
There might be customers who took dependency on this behavior and there are use cases where HTTPS is unnecessary overhead - for example serving public assets.